mlflow/deployments/databricks/__init__.py

```diff
@@ -1,39 +1,135 @@
+        """
         raise NotImplementedError
 
     def delete_deployment(self, name, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def list_deployments(self, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def get_deployment(self, name, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
-    def predict(self, deployment_name=None, inputs=None, endpoint=None):
-        raise NotImplementedError("TODO")
+    def _call_endpoint(
+        self,
+        *,
+        method: str,
+        prefix: str = "/api/2.0",
+        route: Optional[str] = None,
+        json_body: Optional[Dict[str, Any]] = None,
+        timeout: int = MLFLOW_HTTP_REQUEST_TIMEOUT.get(),
+    ):
+        call_kwargs = {}
+        if method.lower() == "get":
+            call_kwargs["params"] = json_body
+        else:
+            call_kwargs["json"] = json_body
+
+        response = http_request(
+            host_creds=get_databricks_host_creds(self.target_uri),
+            endpoint=posixpath.join(prefix, "serving-endpoints", route or ""),
+            method=method,
+            timeout=timeout,
+            raise_on_status=False,
+            retry_codes=MLFLOW_DEPLOYMENT_CLIENT_REQUEST_RETRY_CODES,
+            **call_kwargs,
+        )
+        augmented_raise_for_status(response)
+        return response.json()
 
+    def predict(self, deployment_name=None, inputs=None, endpoint=None):
+        """
+        TODO
+        """
+        return self._call_endpoint(
+            method="POST",
+            prefix="/",
+            route=posixpath.join(endpoint, "invocations"),
+            json_body=inputs,
+            timeout=MLFLOW_DEPLOYMENT_PREDICT_TIMEOUT,
+        )
+
+    # TODO: Return a DatabricksEndpoint object that wraps the response dictionary and exposes some
+    # properties nicely like name and endpoint type (https://github.com/mlflow/mlflow/pull/10425/files#r1396514655)
     def create_endpoint(self, name, config=None):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        config = config.copy() if config else {}
+        payload = {"name": name, "config": config}
+        for key in ("task", "tags"):
+            if val := config.pop(key, None):
+                payload[key] = val
+        return self._call_endpoint(method="POST", json_body=payload)
 
     def update_endpoint(self, endpoint, config=None):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        route = "rate-limits" if "rate_limits" in config else "config"
```

Comment:
Check with Mingyu and sync with databricks APIs

Reply:
@mingyu89 Let me know if there is a better way to support rate limits :)